controllers/memcached/memcached_controller.go

```diff
@@ -256,3 +287,46 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.Service{}).
 		Complete(r)
 }
+
+// GetServerLists returns list of memcached server without/with inet prefix
+func (r *Reconciler) GetServerLists(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *memcachedv1.Memcached,
+) ([]string, []string, error) {
+	var serverList []string
+	var serverListWithPrefix []string
+	var serverPrefix string
+
+	// Define the label selector to filter on
+	labelSelector := map[string]string{
+		"app":            "memcached",
+		"memcached/name": instance.Name,
+	}
+	podList, err := oko_pod.GetPodListWithLabel(ctx, h, instance.Namespace, labelSelector)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if len(podList.Items) == 0 {
+		return nil, nil, fmt.Errorf("No pods could be found")
+	}
```

Comment:
I'm not 100% sure this is an error. It could be a transient state after the statefulset is created but no pods are created yet. Note that if you return error from the reconciliation then a stack trace will be emitted to the log and the reconciliation request is requeued immediately. So this can lead to noisy logs.

Reply:
Alternatively we can let the controller wait for a while with inprogress status, though we don't expect we usually hit this because we run this logic only when at least one replica is available.

Reply:
Fixed in the latest version